waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -420,3 +469,10 @@ getIdFieldFromCrudEntity :: AppSpec -> AS.Crud.Crud -> PslModel.Field
 getIdFieldFromCrudEntity spec crud = fromJust $ Entity.getIdField crudEntity
   where
     crudEntity = snd $ AS.resolveRef spec (AS.Crud.entity crud)
+
+-- | This function assumes that @AppSpec@ it operates on was validated beforehand (with @validateAppSpec@ function).
+-- Example: If user specified their node version range to be [18.2, 20), then this function will return 18.
+getMajorNumberOfLowerBoundOfUserNodeVersionRange :: AppSpec -> Natural
+getMajorNumberOfLowerBoundOfUserNodeVersionRange spec =
+  maybe (error "This should never happen: user node version range lower bound is Inf") SV.major $
+    SVB.versionFromBound $ fst $ SVB.versionBounds $ AS.userNodeVersionRange spec
```

Comment:
I think using the combination of
- `getLowestNodeVersionUserAllows: AppSpec -> Version`
- `major: Version -> Natural`

is a little more readable:
```
"nodeMajorVersion" .= show (major $ getLowestNodeVersionUserAllows $ spec),
```
As opposed to:
```
"nodeMajorVersion" .= show (getMajorNumberOfLowerBoundOfUserNodeVersionRange spec),
```
But both are fine.

Reply:
Makes sense, done!